docs/developing/integrate/middleware/the-graph.md

```diff
@@ -0,0 +1,177 @@
+---
+
+*This page presents a high-level overview of how to create a subgraph for The Graph and the endpoint services provided by Neon EVM.*
+
+## TL;DR
+
+- Work from The Graph CLI
+
+> `yarn global add @graphprotocol/graph-cli` or `npm install -g @graphprotocol/graph-cli`
+
+- Endpoints:
+    - IPFS: https://ipfs.neonevm.org
+    - Graph UI: https://thegraph.neonevm.org
+    - Deployment: https://thegraph.neonevm.org/deploy
+    - GraphQL: https://thegraph.neonevm.org/index-node/graphql
+
+## Prerequisites for standard flow
+
+:::info 
+
+The standard flow is where the end user deploys and collects data from their own contract. In the standard flow you: 
+
+1. Deploy a contract to Neon EVM
+2. Collect the contract's address and block number
+3. [Configure your subgraph.yaml](#subgraph-overview) to collect data on events emitted by contract with value and block number from previous step
+4. [Deploy your subgraph](#deploy-your-subgraph)
+
+ The Graph is highly customizable and supports other flows, for example it allows you to monitor [public contracts](#public-contracts) too. 
+
+::: 
+
+- Smart contract address on Neon EVM 
+- ABI file for the deployed contract 
+- Graph CLI: see [The Graph docs](https://thegraph.com/docs/en/cookbook/quick-start)
+
+## Neon EVM-specific deviations
+
+On the Solana blockchain, data accounts are used to temporarily store data, with a history of days. While on the Ethereum blockchain, data is stored within smart contracts. Neon EVM offers an intermediate approach by storing Solana account data for an extended period to support tracing services.
+
+This means that The Graph may not be able to acquire all data according to the block number assigned. It is important that end users consider what data they wish to extract and store from the API service provided by their subgraph for future use.
+
+## Introduction
+
+To extract, process, and store data from a dApp contract on Neon EVM using The Graph protocol, you must deploy a dedicated subgraph to a Graph node. 
+
+:::info
+Subgraphs map 1:1 with a dApp to provide Graph nodes with the information and logic needed to:
+- Observe the blockchain for log events of smart contacts
+- Translate the events into entities for storage
+
+:::
+
+### Subgraph overview
+
+We need to create a subgraph to source data from the contract/s of interest that will constantly observe the blockchain using [your chosen Neon RPC](https://chainlist.org/?chain=245022926&testnets=true&search=Neon+EVM) for your events on interest. When one of these events is detected, the Graph node will extract the event log data and begin to process the information using a WebAssembly script defined in the subgraph.
+
+The script uses a GraphQL schema file in the subgraph to produce records, called entities, that represent metadata related to the recently created or updated Gravatar. These entities are stored on a database so they may be queried by API requests.
```

Comment:
What `Gravatar` do you reffer to? Didn't catch the context here

Reply:
nice catch @romanova-natasha -- I repurposed a tutorial, this is defunct!